### PR TITLE
CORTX_31103_test_pod_failure

### PR DESCRIPTION
**Pull request by ankita-93**
Signed-off-by: ankita-93 <ankita.thummar@seagate.com>

# Problem Statement
- Test update under [https://jts.seagate.com/browse/CORTX-31101](url)

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/test_pod_failure.py [hunk 1]
```diff
@@ -61,7 +60,7 @@
 # pylint: disable=R0904
 class TestPodFailure:
```
Comment (ApurwaMohite) [inline tests/ha/test_pod_failure.py:61]: We can add Data in class name also.

File: tests/ha/test_pod_failure.py [hunk 2]
```diff
@@ -2106,7 +1898,7 @@ def test_mpu_during_pod_unsafe_shutdown(self):
         LOGGER.info("Step 2: Shutdown the data pod by deleting deployment (unsafe)")
         LOGGER.info("Get pod name to be deleted")
         pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
-        pod_name = random.sample(pod_list, 1)[0]
+        pod_name = self.system_random.sample(pod_list, 1)[0]
```
Comment (ApurwaMohite) [inline tests/ha/test_pod_failure.py:1901]: Why pod deletion is not optimized in this test?

Comment (ankita-93) [reply tests/ha/test_pod_failure.py:1901]: There is one more PR i will be raising for during tests

File: tests/ha/test_pod_failure.py [hunk 3]
```diff
@@ -910,10 +773,9 @@ def test_pod_shutdown_delete_deployment(self):
     @CTFailOn(error_handler)
     def test_deletes_during_pod_down(self):
         """
-        This test tests continuous DELETEs during data pod down by deleting deployment
+        This test tests DELETEs during data pod down by deleting deployment
         """
-        LOGGER.info("STARTED: Test to verify continuous DELETEs during data pod down by "
-                    "deleting deployment.")
+        LOGGER.info("STARTED: Test to verify DELETEs during data pod down by deleting deployment.")
         event = threading.Event()  # Event to be used to send intimation of pod deletion
```
Comment (rahul-hatwar) [inline tests/ha/test_pod_failure.py:779]: data pod deleteion

File: tests/ha/test_pod_failure.py [hunk 4]
```diff
@@ -1226,13 +1067,13 @@ def test_pod_shutdown_scale_replicas(self):
...
     def test_reads_writes_during_pod_down(self):
         """
-        This test tests Continuous READs and WRITEs during data pod down (delete deployment)
+        This test tests READs and WRITEs during data pod down (delete deployment)
         """
-        LOGGER.info("STARTED: Test to verify Continuous READs and WRITEs during data pod down by "
-                    "delete deployment.")
+        LOGGER.info("STARTED: Test to verify READs and WRITEs during data pod down by delete "
+                    "deployment.")
         event = threading.Event()  # Event to be used to send intimation of pod deletion
```
Comment (rahul-hatwar) [inline tests/ha/test_pod_failure.py:1074]: data pod deletion, check this complete file

**Review by rahul-hatwar — APPROVED** [review]
minor comments otherwise LGTM

**Review by ApurwaMohite — APPROVED** [review]

**Comment by ankita-93** [comment]
image
@sonalk0209 @ApurwaMohite  collection logs are clean

**Review by sonalk0209 — APPROVED** [review]

**Pull request merged**